bot/exts/moderation/stream.py

```diff
@@ -0,0 +1,149 @@
+                try:
+                    member = await self.bot.get_guild(Guild.id).fetch_member(key)
+                except discord.errors.NotFound:
+                    log.debug(
+                        f"Member {key} left the guild before we could scheudle "
+                        "the revoking of their streaming permissions."
+                    )
+                    await self.task_cache.delete(key)
+                    continue
+                except discord.HTTPException as e:
+                    log.exception(f"Exception while trying to retrieve member {key} from discord\n{e}")
+                    continue
+            revoke_time = datetime.datetime.utcfromtimestamp(value)
+            log.debug(f"Scheduling {member} ({member.id}) to have streaming permission revoked at {revoke_time}")
+            self.scheduler.schedule_at(
+                revoke_time,
+                key,
+                self._remove_streaming_permission(member)
+            )
+
+    @commands.command(aliases=("streaming",))
+    @commands.has_any_role(*STAFF_ROLES)
+    async def stream(self, ctx: commands.Context, user: discord.Member, duration: Expiry = None) -> None:
+        """
+        Temporarily grant streaming permissions to a user for a given duration.
+
+        A unit of time should be appended to the duration.
+        Units (∗case-sensitive):
+        \u2003`y` - years
+        \u2003`m` - months∗
+        \u2003`w` - weeks
+        \u2003`d` - days
+        \u2003`h` - hours
+        \u2003`M` - minutes∗
+        \u2003`s` - seconds
+
+        Alternatively, an ISO 8601 timestamp can be provided for the duration.
+        """
+        log.trace(f"Attempting to give temporary streaming permission to {user} ({user.id}).")
+        # if duration is none then calculate default duration
+        if duration is None:
+            now = datetime.datetime.utcnow()
+            duration = now + datetime.timedelta(minutes=VideoPermission.default_permission_duration)
+
+        # Check if user already has streaming permission
+        already_allowed = any(Roles.video == role.id for role in user.roles)
+        if already_allowed:
+            await ctx.send(f"{Emojis.cross_mark} This user can already stream.")
+            log.debug(f"{user} ({user.id}) already has permission to stream.")
+            return
+
+        # Schedule task to remove streaming permission from Member and add it to task cache
+        self.scheduler.schedule_at(duration, user.id, self._remove_streaming_permission(user))
+        await self.task_cache.set(user.id, duration.timestamp())
+        await user.add_roles(discord.Object(Roles.video), reason="Temporary streaming access granted")
+        revoke_time = format_infraction_with_duration(str(duration))
+        await ctx.send(f"{Emojis.check_mark} {user.mention} can now stream until {revoke_time}.")
+        log.debug(f"Successfully given {user} ({user.id}) permission to stream until {revoke_time}.")
```

Comment:
There are several places in this file that have these clusters of lines. Adding some blank lines between different "sections" would help with readability.